hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractInvocationFuture.java

```diff
@@ -262,10 +262,12 @@ public void run() {
 
             });
         } catch (RejectedExecutionException e) {
-            callback.onFailure(e);
+            callback.onFailure(wrapToInstanceNotActiveException(e));
```

Comment:
Can the executor be user provided? (e.g. ICompletableFuture.andThen user can provide the executor) User executor returning this exception may not always mean that the client is being shutdown, does it?

Reply:
@ihsandemir I send a second commit to address your concern. It makes sense to keep `RejectedExecutionException` when the user provides the executor.